File: backend/src/main/java/com/magesty/backend/models/Adresse.java
package com.magesty.backend.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.magesty.backend.models.dto.AdresseDto;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;
import java.util.UUID;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
public class Adresse implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long _id;
    private String avenue;
    private String pays;
    private String ville;
    private String phone;
    private String email;

    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "adresseId")
    private Admin admin;

    public static Adresse from(AdresseDto adresseDto){
        Adresse adresse = new Adresse();

        if(Objects.isNull(adresseDto)){
            return null;
        }else{
            adresse.set_id(adresseDto.getId());
            adresse.setAvenue(adresseDto.getAvenue());
            adresse.setEmail(adresseDto.getEmail());
            adresse.setPhone(adresseDto.getPhone());
            adresse.setPays(adresseDto.getPays());
            adresse.setVille(adresseDto.getVille());

            return adresse;
        }
    }


}

File: frontend-ng/src/app/models/Adresse/adresse.ts
export class Adresse {
  private _id: number;
  private _avenue: string;
  private _pays: string;
  private _ville: string;
  private _phone: string;
  private _email: string;

  constructor(id: number, avenue: string, pays: string, ville: string, phone: string, email: string) {
    this._id = id;
    this._avenue = avenue;
    this._pays = pays;
    this._ville = ville;
    this._phone = phone;
    this._email = email;
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get avenue(): string {
    return this._avenue;
  }

  set avenue(value: string) {
    this._avenue = value;
  }

  get pays(): string {
    return this._pays;
  }

  set pays(value: string) {
    this._pays = value;
  }

  get ville(): string {
    return this._ville;
  }

  set ville(value: string) {
    this._ville = value;
  }

  get phone(): string {
    return this._phone;
  }

  set phone(value: string) {
    this._phone = value;
  }

  get email(): string {
    return this._email;
  }

  set email(value: string) {
    this._email = value;
  }
}
